framework/src/base/MooseApp.C

```diff
@@ -458,7 +458,7 @@ MooseApp::MooseApp(InputParameters parameters)
   _the_warehouse->registerAttribute<AttribPostAux>("post_aux", 0);
   _the_warehouse->registerAttribute<AttribName>("name", "dummy");
   _the_warehouse->registerAttribute<AttribSystem>("system", "dummy");
-  _the_warehouse->registerAttribute<AttribVar>("variable", 0);
+  _the_warehouse->registerAttribute<AttribVar>("variable", -1);
```

Comment:
0 is actually a value that a variable number can take on, in fact it will be the value that the first variable always takes on, so it's actually the worst possible default attribute value you could give. It's been so long since I ran into this issue but I remember this default value caused me some real debugging hell, maybe a full day's worth. By changing to -1, which is a default value that no actual variable will ever have, this becomes safe.